src/lib/y2partitioner/dialogs/bcache.rb

```diff
@@ -199,6 +215,60 @@ def store
         # Otherwise undefined
         attr_reader :result
       end
+
+      # Widget to select the cache mode
+      class CacheMode < CWM::ComboBox
+        # @param device [Y2Storage::Bcache,nil] existing caching device or nil if it is new bcache
+        def initialize(device)
+          textdomain "storage"
+          @device = device
+        end
+
+        # @macro seeAbstractWidget
+        def label
+          _("Cache Mode")
+        end
+
+        # @macro seeItemsSelection
+        def items
+          Y2Storage::CacheMode.all.map do |mode|
+            [mode.to_sym.to_s, mode.to_human_string]
+          end
+        end
+
+        # @macro seeAbstractWidget
+        # @see https://en.wikipedia.org/wiki/Cache_(computing)#Writing_policies
+        def help
+          # TRANSLATORS: %s stands for name of option
+          format(_(
+                   "<b>%s</b> is the operating mode for bcache. " \
+                   "There are currently four supported modes. " \
+                   "<i>Writethrough</i> is default mode which caches reading and do write parallel. " \
+                   "So in case of failure of caching device, no data is lost." \
+                   "<i>Writeback</i> is mode that caches also writes. This result in better " \
+                   "performance of write operations." \
+                   "<i>Writearound</i> is mode that use cache only for reads and writes new " \
+                   "content only to backing device." \
+                   "<i>None</i> is mode that do not use cache for read neither for write. " \
+                   "Useful mainly for temporary disabling cache before big sequential read/write " \
+                   "which otherwise result in intensive overwritting of cache device."
+          ), label)
```

Comment:
Before entering into idiomatic suggestions, I miss some formatting here. The help of this dialog looks ugly as hell:

![help-unformatted](https://user-images.githubusercontent.com/3638289/47213361-0c075980-d39b-11e8-9eca-a246914d4b33.png)

Reply:
Btw, while I'm reading the whole thing. It says "the caching device will be formatted so any previous content will be wiped out". Is that also true if the selected caching device is an existing bcache cset? Just curious

Reply:
what version of CWM you use? It looks like bug of CWM itself that it does not separate individual entries. It should be separated by <p> each option.

Reply:
I can try how ul will look for cache mode itself.

Reply:
And now, my suggestions about wording. Nothing too big.

* _Writethrough_ reading operations are cached, writing is done in parallel to both devices. In case of failure of the caching device, no data is lost. This is the default mode.
* _Writeback_ both reading and writing operations are cached. This result in better performance when writing.
* _Writearound_ reading is cached, new content is written only to the backing device.
* _None_ disables all caching both for reading and writing. This is useful mainly for temporarily disabling the cache before any big sequential read or write, otherwise that would result in intensive overwriting on the cache device.

Reply:
I tune a bit to something like, even list looks good:

![snimek obrazovky_2018-10-19_12-56-53](https://user-images.githubusercontent.com/478871/47214536-85547b80-d39e-11e8-9029-f4ba304f74bf.png)

Reply:
> Btw, while I'm reading the whole thing. It says "the caching device will be formatted so any previous content will be wiped out". Is that also true if the selected caching device is an existing bcache cset? Just curious

if it is existing bcache cset, then it is just attached and not created. So not wiped out.